### PR TITLE
sanitize names coming from openapi better

### PR DESCRIPTION
**Pull request by lewisjkl**
ShapeIds can only contain certain characters and still be valid. As such, this PR makes it so we will never create ShapeIds that contain invalid characters.

### CODE REVIEW

File: modules/openapi/tests/src/StructureSpec.scala [hunk 1]
```diff
@@ -384,4 +384,29 @@ final class StructureSpec extends munit.FunSuite {
...
+                     |      type: object
+                     |      properties:
+                     |        version3.1:
+                     |          type: string
+                     |""".stripMargin
+
+    val expectedString = """|namespace foo
+                      |
+                      |structure Object {
+                      | version31: String
```
Comment (kubukoz) [inline modules/openapi/tests/src/StructureSpec.scala:405]: should this have a `@jsonName("version3.1")`?

Comment (lewisjkl) [reply modules/openapi/tests/src/StructureSpec.scala:405]: Ah yeah, I totally overlooked that. Thanks will add

Comment (kubukoz) [reply modules/openapi/tests/src/StructureSpec.scala:405]: also worth checking if this affects unions / other places that `@jsonName` is viable in :)

Comment (lewisjkl) [reply modules/openapi/tests/src/StructureSpec.scala:406]: Added for structures in https://github.com/disneystreaming/smithy-translate/pull/48/commits/49fb2075405e102312705bce73dd8ee8c6878bbd will look at unions now

Comment (lewisjkl) [reply modules/openapi/tests/src/StructureSpec.scala:406]: Okay and done for unions in https://github.com/disneystreaming/smithy-translate/pull/48/commits/c171da9b107b8050b600770575946fb1c7d08951

Comment (kubukoz) [reply modules/openapi/tests/src/StructureSpec.scala:406]: neato!

**Review by yisraelU — APPROVED** [review]

**Pull request merged**